parseFunctions.cpp

```diff
@@ -3,294 +3,279 @@
+	     token=="~"||token=="*"||token=="/"||token=="%"||token=="ALL"||token=="ANY"||
+	     token=="BETWEEN"||token=="EXISTS"||token=="LIKE"||token=="SOME"||token=="UNION"||
+	     token=="INTERSECT"||token=="EXCEPT"||token=="AVG"||token=="COUNT"||token=="SUM"||
+	     token=="<"||token==">"||token=="="||token==">="||token=="<="||token=="<>"||
+	     token=="!="||token=="!>"||token=="!<")
+	    return true;
+	  else
+	    return false;
 	}
-	
-	return valid;
-}
 
-/*	validates that all attributes are present in the schema
- * 	returns true if attribute or attribute list is valid
- */
-bool parse_attributes() {
-	string wholeToken = "";
-	bool valid = false;
-	
-	// checks if there is a list of attributes
-	if(token.find(".") < strlen(token.c_str())) {		// check if attribute includes a '.'
-		schemaLL *runner = &schema;
-		string table = token;
+	bool isInteger(){
+	  int i=0, k=0, tok_size, state=1;
+	  char tok[50];
+
+	  strncpy(tok, token.c_str(), sizeof(tok)); //turn token (string) into chars, store in char array c[].
+	  tok[sizeof(tok)-1] = 0;
+	  tok_size = sizeof(tok);  
+
+	  while(k<tok_size){
+	    switch(state){
+	      case 1:
+		if(tok[k]=='+' || tok[k]=='-')
+		  state=2;
+		else if(tok[k]>='0' || tok[k]<='9')
+		  state=3;
+		else
+		  state=4;
+	      break;
+	      case 2:
+		if(tok[k]>='0' || tok[k]<='9')
+		  state=3;
+		else
+		  state=4;
+	      break;
+	      case 3:
+		if(tok[k]>='0' || tok[k]<='9')
+		  state=3;
+		else
+		  state=4;
+	      break;
+	      default:        //if in state 4, failure
+		return false;
+	    }//switch
+	  }//while
+	  if(k==tok_size){
+	    switch(state){
+	      case 3: return true;
+	      default: return false;
+	    }//switch
+	  }//if
+	}
 
-		table.erase(table.find("."), strlen(table.c_str()));	// get table associated with attribute
-		token.erase(0, token.find(".") + 1);					// get attribute
+	bool isAlias(){
+	    int i=0, k=0, tok_size, state=1;
+	  char tok[50];
+
+	  strncpy(tok, token.c_str(), sizeof(tok)); //turn token (string) into chars, store in char array c[].
+	  tok[sizeof(tok)-1] = 0;
+	  tok_size = sizeof(tok);
+
+	  if(!isKeyword(token)){
+	    while(k<tok_size){ //for each character in tok
+	      switch(state){
+		case 1:
+		  if(tok[k]>='a' && tok[k]<='z')
+		    state=3;
+		  else
+		    state=2;
+		  break;
+		case 3:
+		  if((tok[k]>='a' && tok[k]<='z')||(tok[k]>='0' && tok[k]<='9')
+		    state=3;
+		  else
+		    state=2;
+		  break;
+		default: //state 2=failure
+		  return false;
+	      }//switch
+	      k++;
+	    }//while
+	    if(k==tok_size){
+	      switch(state){
+		case 3: return true;
+		default: return false;
+	      }//switch
+	    }//if
+	  }
+	}
 
-		while(runner != NULL) {
-			if(table == runner->m_tableName) {
-				if(token == runner->m_attributeName) {
-					valid = true;
-					break;
-				}
-				if(token == runner->m_attributeName + ",") {
-					getToken();
-					valid = parse_attributes();
-					break;
-				}
-			}
-			runner = runner->m_next;
```

Comment:
Why don't we keep this while loop or something similar instead of hardcoding in the table name? Same with the attributes